### PR TITLE
Crowdstrike search endpoints by hash

### PR DESCRIPTION
**Pull request by karinafishman**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-5565

## Description
1. Playbook input only show 'sha256' but can handle all IOC types.
2. The playbook fails when there is no input and 'sha256' argument is not mandatory.
3. Added the playbook to "Search Endpoints By Hash - Generic V2" playbook.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by ssokolovich — APPROVED** [review]

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml [hunk 1]
```diff
@@ -34,16 +35,15 @@ tasks:
...
   "1":
     id: "1"
     taskid: 9ce410b8-ddde-4690-8625-2cfab080cd83
     type: regular
     task:
       id: 9ce410b8-ddde-4690-8625-2cfab080cd83
       version: -1
-      name: Search for hash in enviorment
-      description: Returns a list of device IDs an indicator ran on.
+      name: Get device id list by SHA1
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml:45]:
```suggestion
      name: Get device ID list by SHA1
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml [hunk 2]
```diff
@@ -246,19 +288,570 @@ tasks:
...
+    isautoswitchedtoquietmode: false
+  "13":
+    id: "13"
+    taskid: b930b6ef-bb8c-417e-88a8-d5cecabb228d
+    type: condition
+    task:
+      id: b930b6ef-bb8c-417e-88a8-d5cecabb228d
+      version: -1
+      name: Is there any MD5?
+      description: Is there any MD5 provided?
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml:494]:
```suggestion
      description: Are there any MD5 files provided?
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml [hunk 3]
```diff
@@ -246,19 +288,570 @@ tasks:
...
+    isautoswitchedtoquietmode: false
+  "14":
+    id: "14"
+    taskid: 169383f7-4adc-460a-8cf3-125b8422fecc
+    type: condition
+    task:
+      id: 169383f7-4adc-460a-8cf3-125b8422fecc
+      version: -1
+      name: Is there any SHA1?
+      description: Is there any SHA1 provided?
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml:540]:
```suggestion
      description: Are there any SHA1 files provided?
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml [hunk 4]
```diff
@@ -246,19 +288,570 @@ tasks:
...
+    isautoswitchedtoquietmode: false
+  "15":
+    id: "15"
+    taskid: 2334a9cf-94c0-4b2c-81fc-90330616c546
+    type: condition
+    task:
+      id: 2334a9cf-94c0-4b2c-81fc-90330616c546
+      version: -1
+      name: Is there any SHA256?
+      description: 'Is there any SHA256 provided '
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml:586]:
```suggestion
      description: Are there any SHA256 files provided?
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml [hunk 5]
```diff
@@ -246,19 +288,570 @@ tasks:
...
+    isautoswitchedtoquietmode: false
+  "18":
+    id: "18"
+    taskid: a69c0320-8da4-4f7d-8c06-e1b3733e544f
+    type: condition
+    task:
+      id: a69c0320-8da4-4f7d-8c06-e1b3733e544f
+      version: -1
+      name: Should enrich the returned results?
+      description: "If the hash was found on additional hosts the playbook will enrich the endpoint details.\nIn case we provided host id and there is no additional endpoints that were found, the playbook will be will stop and finish without the enrich. "
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml:712]:
```suggestion
      description: "If the hash was found on additional hosts the playbook will enrich the endpoint details.\nIn case we provided a host ID and there is no additional endpoints that were found, the playbook will stop and finish without the enrichment."
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml [hunk 6]
```diff
@@ -246,19 +288,570 @@ tasks:
...
+    isautoswitchedtoquietmode: false
+  "19":
+    id: "19"
+    taskid: d61e2bb3-62ba-4eaa-850f-e4c4e0ab3342
+    type: condition
+    task:
+      id: d61e2bb3-62ba-4eaa-850f-e4c4e0ab3342
+      version: -1
+      name: Should enrich the returned results?
+      description: "If the hash was found on additional hosts the playbook will enrich the endpoint details.\nIn case we provided host id and there is no additional endpoints that were found, the playbook will be will stop and finish without the enrich. "
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash.yml:792]:
```suggestion
      description: "If the hash was found on additional hosts the playbook will enrich the endpoint details.\nIn case we provided a host ID and there is no additional endpoints that were found, the playbook will stop and finish without the enrichment. "
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash_README.md [hunk 7]
```diff
@@ -1,40 +1,58 @@
...
 This playbook uses the following sub-playbooks, integrations, and scripts.
 
 ### Sub-playbooks
+
 This playbook does not use any sub-playbooks.
 
 ### Integrations
-CrowdStrikeFalcon
+
+* CrowdStrikeFalcon
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash_README.md:14]:
```suggestion
CrowdStrikeFalcon
```

File: Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash_README.md [hunk 8]
```diff
@@ -1,40 +1,58 @@
...
 
 ### Scripts
-This playbook does not use any scripts.
+
+* IsIntegrationAvailable
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/Playbooks/playbook-CrowdStrike_Falcon_-_Search_Endpoints_By_Hash_README.md:18]:
```suggestion
IsIntegrationAvailable
```

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_17.md [hunk 9]
```diff
@@ -0,0 +1,8 @@
+
+#### Playbooks
+
+##### CrowdStrike Falcon - Search Endpoints By Hash
+
+- Added support for additional hash types: MD5 and SHA1.
+- Added task to check that the inputs were provided.
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_17.md:7]:
```suggestion
- Added a task to check that the inputs were provided.
```

File: Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_17.md [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+- Edited the condition of the enrich endpoints in order to be able to use this playbook as part of "Malware Investigation and Response" and as part of "Search Endpoints by hash - Generic".
```
Comment (ShirleyDenkberg) [inline Packs/CrowdStrikeFalcon/ReleaseNotes/1_12_17.md:8]:
```suggestion
- Edited the condition of the enriched endpoints in order to be able to use this playbook as part of "Malware Investigation and Response" and as part of "Search Endpoints by hash - Generic".
```

**Comment by ShirleyDenkberg** [comment]
@ssokolovich Doc review completed.

**Pull request merged**